bundles/org.eclipse.core.resources/src/org/eclipse/core/internal/events/ResourceChangeListenerList.java

```diff
@@ -79,29 +73,23 @@ public String toString() {
 	 * @param mask event types
 	 */
 	public synchronized void add(IResourceChangeListener listener, int mask) {
-		Assert.isNotNull(listener);
 		if (mask == 0) {
 			remove(listener);
 			return;
 		}
 		ResourceChangeListenerList.ListenerEntry entry = new ResourceChangeListenerList.ListenerEntry(listener, mask);
-		final int oldSize = listeners.length;
+		final int oldSize = listeners.size();
 		// check for duplicates using identity
 		for (int i = 0; i < oldSize; ++i) {
-			if (listeners[i].listener == listener) {
-				removing(listeners[i].eventMask);
+			if (listeners.get(i).listener == listener) {
+				removing(listeners.get(i).eventMask);
```

Comment:
This is interesting (unrelated to the current change). Javadoc says no effect, but here if the listener was added previously to listen with different mask, the mask will be changed.